From c3204ab86199364b34892a346431a727e4a84b22 Mon Sep 17 00:00:00 2001
From: Rashed Abdel-Tawab <rashed@linux.com>
Date: Wed, 18 Oct 2017 19:53:33 -0400
Subject: [PATCH 1/3] lineage-sdk: Import power menu related classes

Different parts of the system need to check if advanced reboot is enabled,
so just move the method here and include it in said different parts.

The global actions constants also used to live in the frameworks, but as
they include our own additions and are also accessible by LineageParts,
place them here from now on.

Change-Id: Ic5a02b8118c702dced5a25d775a4cc84c92a3fc2
---
 .../internal/util/PowerMenuConstants.java          | 51 ++++++++++++++++++++++
 .../org/lineageos/internal/util/RebootUtils.java   | 35 +++++++++++++++
 2 files changed, 86 insertions(+)
 create mode 100644 sdk/src/java/org/lineageos/internal/util/PowerMenuConstants.java
 create mode 100644 sdk/src/java/org/lineageos/internal/util/RebootUtils.java

diff --git a/sdk/src/java/org/lineageos/internal/util/PowerMenuConstants.java b/sdk/src/java/org/lineageos/internal/util/PowerMenuConstants.java
new file mode 100644
index 0000000..a3e2469
--- /dev/null
+++ b/sdk/src/java/org/lineageos/internal/util/PowerMenuConstants.java
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2015 The CyanogenMod Project
+ * Copyright (C) 2017 The LineageOS Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.lineageos.internal.util;
+
+/* Master list of all actions for the power menu */
+public class PowerMenuConstants {
+    public static final String GLOBAL_ACTION_KEY_POWER = "power";
+    public static final String GLOBAL_ACTION_KEY_RESTART = "restart";
+    public static final String GLOBAL_ACTION_KEY_SCREENSHOT = "screenshot";
+    public static final String GLOBAL_ACTION_KEY_AIRPLANE = "airplane";
+    public static final String GLOBAL_ACTION_KEY_USERS = "users";
+    public static final String GLOBAL_ACTION_KEY_SETTINGS = "settings";
+    public static final String GLOBAL_ACTION_KEY_LOCKDOWN = "lockdown";
+    public static final String GLOBAL_ACTION_KEY_BUGREPORT = "bugreport";
+    public static final String GLOBAL_ACTION_KEY_SILENT = "silent";
+    public static final String GLOBAL_ACTION_KEY_VOICEASSIST = "voiceassist";
+    public static final String GLOBAL_ACTION_KEY_ASSIST = "assist";
+
+    private static String[] ALL_ACTIONS = {
+        GLOBAL_ACTION_KEY_POWER,
+        GLOBAL_ACTION_KEY_RESTART,
+        GLOBAL_ACTION_KEY_SCREENSHOT,
+        GLOBAL_ACTION_KEY_AIRPLANE,
+        GLOBAL_ACTION_KEY_USERS,
+        GLOBAL_ACTION_KEY_SETTINGS,
+        GLOBAL_ACTION_KEY_LOCKDOWN,
+        GLOBAL_ACTION_KEY_BUGREPORT,
+        GLOBAL_ACTION_KEY_SILENT,
+        GLOBAL_ACTION_KEY_VOICEASSIST,
+        GLOBAL_ACTION_KEY_ASSIST
+    };
+
+    public static String[] getAllActions() {
+        return ALL_ACTIONS;
+    }
+}
diff --git a/sdk/src/java/org/lineageos/internal/util/RebootUtils.java b/sdk/src/java/org/lineageos/internal/util/RebootUtils.java
new file mode 100644
index 0000000..ca0c925
--- /dev/null
+++ b/sdk/src/java/org/lineageos/internal/util/RebootUtils.java
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2017 The LineageOS Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.lineageos.internal.util;
+
+import android.app.KeyguardManager;
+import android.content.Context;
+import android.os.UserHandle;
+
+import lineageos.providers.LineageSettings;
+
+public final class RebootUtils {
+    public static boolean isAdvancedRebootPossible(final Context context) {
+        KeyguardManager km = (KeyguardManager) context.getSystemService(Context.KEYGUARD_SERVICE);
+        boolean keyguardLocked = km.inKeyguardRestrictedInputMode() && km.isKeyguardSecure();
+        boolean advancedRebootEnabled = LineageSettings.Secure.getInt(context.getContentResolver(),
+                LineageSettings.Secure.ADVANCED_REBOOT, 0) == 1;
+        boolean isPrimaryUser = UserHandle.getCallingUserId() == UserHandle.USER_OWNER;
+
+        return advancedRebootEnabled && !keyguardLocked && isPrimaryUser;
+    }
+}
-- 
2.7.4

